Always define the path DNSSEC_OPENSSL_CONF

flo-renaud opened the pull request:
The variable was None by default and set to /etc/ipa/dnssec/openssl.cnf
for fedora only because the code is specific to the support of pkcs11
engine for bind. As a consequence ipa-backup had a "None" value in the
list of files to backup and failed on Exception.

ipa-backup code is able to handle missing files, and the code using
the pkcs11 engine is called only when NAMED_OPENSSL_ENGINE is set
(only in fedora so far). It is safe to always define a value for
DNSSEC_OPENSSL_CONF even on os where it does not exist.

Fixes: https://pagure.io/freeipa/issue/8597

flo-renaud commented:
Only ipa-4-9 label is needed as the code was committed to the master branch only, and ipa-4-9 inherited from the issue as it branched after the commit.

stanislavlevin commented:
Hi, @flo-renaud. Thank's for the fix.
I think it would be better to handle `None`, as paths can be so.
For example, with:
```
list(filter(None, files))
```
and similar for dirs and logs.

abbra commented:
I think we need both. ;) For example, Fedora ELN (Rawhide right now = RHEL 9 base) already has this and would need to have support for openssl engine. Also, RHEL 8.4 gets all the same bits, including bind-dyndb-ldap, so it can do the same if we want.

flo-renaud commented:
@stanislavlevin thanks for your suggestion. I added a check for non-None item in verify_directories, that should protect us from other constants set to None. Also added a temp commit to launch ipa-backup tests.

abbra commented:
LGTM. Please remove the temporary commit.

flo-renaud commented:
@abbra Temp commit removed and tests are green.

flo-renaud commented:
master:

* 3b007b7bba185989bec34bb071742270d3362e2e Always define the path DNSSEC_OPENSSL_CONF
